flowauth/backend/flowauth/user_settings.py

```diff
@@ -225,6 +227,18 @@ def is_two_factor_active():
     )
 
 
+@blueprint.route("/roles/server/<server_id>")
+@login_required
+def get_user_roles(server_id):
+    """
+    Returns a list of roles on server_id for the logged in user
+    """
+    server = Server.query.filter(Server.id == server_id).first_or_404()
+    roles = Role.query.filter(Role.server == server).join(User.roles).all()
+    # NOTE: This includes other users inside the role, leading to enumeration. Is this a problem?
```

Comment:
It doesn't sound clean but others may have a better idea.
Can you not filter the other users out?